### PR TITLE
Set PDEATHSIG to SIGKILL on child processes after reexec

### LINKED ISSUE #3345: Teleport sometimes creates orphan shell processes.
A customer has reported that if their session is interrupted by network glitch or client crash bash session on the remote machine is not closed. For example:

```
init(1)─┬─abrt-dump-oops(2826)
        ├─acpid(2112)
        ├─agetty(2947)
        ├─atd(2876)
        ├─bash(2937)───sleep(29307)
        ├─bash(25414,username)
```

Note `bash(25414,username)`.

This is most likely due to the chances in shell execution introduced in Teleport 4.2. Teleport now reexecs itself before launching a shell. If the parent process dies, the children will be re-attached to PID 1.

The fix is most likely going to be to create a process group during execution and send a kill signal to all.

### PR DESCRIPTION
**Pull request by awly**
To execute an SSH command, Teleport re-executes itself and execs the
command from this child process:
  `teleport -> teleport exec -> sh -c "user command"`

Both parent teleport processes could exit unexpectedly (from `SIGKILL` or
even connection interruption).

Make sure all child processes get cleaned up and not orphaned to PID 1:
- teleport exec via `SIGQUIT` to request [graceful shutdown](https://github.com/gravitational/teleport/blob/69f069863619c624c5dc260638703b3429866d87/lib/service/signals.go#L80-L84)
- user command via `SIGKILL` because it might ignore other signals

### CODE REVIEW

**Comment by awly** [comment]
@gravitational-jenkins retest this please

**Review by klizhentas — APPROVED** [review]

**Comment by awly** [comment]
Will wait for @russjones review before merging

**Review by russjones — APPROVED** [review]

**Review by fspmarshall — APPROVED** [review]
Note: This breaks compilation when targeting non-linux platforms.  We don't compile `lib/srv` for non-linux platforms currently (AFAIK), so this is fine.  If we need to change this in the future its a fairly easy fix with either build flags or use of the `reflect` package.

**Comment by awly** [comment]
retest this please

**Comment by awly** [comment]
retest this please

**Comment by awly** [comment]
retest this please

**Comment by awly** [comment]
retest this please

**Comment by awly** [comment]
retest this please

**Comment by webvictim** [comment]
We do compile the `teleport` binary for MacOS; will it work there?

**Comment by webvictim** [comment]
@awly Also as a general usability thing, could you raise branches against `gravitational/teleport` rather than your own fork? I'm trying to check this out to see whether I can compile it on MacOS and the branch isn't present in the gravitational repo, so I have to do some fiddling with remotes etc.

**Comment by awly** [comment]
Ah, didn't realize `teleport` was built for darwin. None of the make rules caught that (#3594).
@webvictim can you link the build script for that?

> @awly Also as a general usability thing, could you raise branches against gravitational/teleport rather than your own fork?

Yeah, I switched to that after my initial few PRs. This is the last PR of that initial batch that I did on my fork.

**Comment by awly** [comment]
retest this please

**Comment by webvictim** [comment]
The build script is here: https://github.com/gravitational/ops/blob/master/xcloud/release-teleport.sh

It gets run by Ansible (https://github.com/gravitational/ops/blob/master/xcloud/release-teleport.yaml) which is called from Jenkins based on tags (https://jenkins.gravitational.io/job/TeleportPublish/427/parameters/)

**Pull request merged**
